planner/cascades/implementation_rules.go

```diff
@@ -151,9 +154,35 @@ func (r *ImplShow) OnImplement(expr *memo.GroupExpr, reqProp *property.PhysicalP
 
 	// TODO(zz-jason): unifying LogicalShow and PhysicalShow to a single
 	// struct. So that we don't need to create a new PhysicalShow object, which
-	// can help us to reduce the gc presure of golang runtime and improve the
+	// can help us to reduce the gc pressure of golang runtime and improve the
 	// overall performance.
 	showPhys := plannercore.PhysicalShow{ShowContents: show.ShowContents}.Init(show.SCtx())
 	showPhys.SetSchema(logicProp.Schema)
 	return impl.NewShowImpl(showPhys), nil
 }
+
+// ImplSelection is the implementation rule which implements LogicalSelection
+// to PhysicalSelection.
+type ImplSelection struct {
+}
+
+// Match implements ImplementationRule Match interface.
+func (r *ImplSelection) Match(expr *memo.GroupExpr, prop *property.PhysicalProperty) (matched bool) {
+	return true
+}
+
+// OnImplement implements ImplementationRule OnImplement interface.
+func (r *ImplSelection) OnImplement(expr *memo.GroupExpr, reqProp *property.PhysicalProperty) (memo.Implementation, error) {
+	logicalSel := expr.ExprNode.(*plannercore.LogicalSelection)
+	physicalSel := plannercore.PhysicalSelection{
+		Conditions: logicalSel.Conditions,
+	}.Init(logicalSel.SCtx(), expr.Group.Prop.Stats.ScaleByExpectCnt(reqProp.ExpectedCnt), logicalSel.SelectBlockOffset(), reqProp)
```

Comment:
Sure. I just find another problem. 
The `Stats` for `init` should be `expr.Group.Prop.Stats` instead of `expr.Group.Prop.Stats.ScaleByExpectCnt(reqProp.ExpectedCnt)`?

Reply:
And `reqProp.ExpectedCnt` also should be scaled by the selectivity?